lib/App/DuckPAN/Web.pm

```diff
@@ -285,11 +285,17 @@ sub request {
 					# Inject a script which prints out what we want.
 					# There is no error-checking or support for non-auto-templates here.
 					my $structured = $result->structured_answer;
-					my $template_name = 'goodie_'.scalar @{$structured->{input}}.'_inputs';
-					my $json_string = encode_json({Answer => $structured});
-					$zci_container->push_content(HTML::TreeBuilder->new_from_content("<script>\$(window).load(function(){"
-								. "document.getElementById('zci-answer').innerHTML = DDG.exec_template('$template_name', $json_string);"
-								. "});</script>")->guts);
+					if(exists $structured->{templates}){ # user-specified templates
+						push @calls_goodie, $structured;
+						last;
+					}
+					else{ # auto-template
```

Comment:
@zachthompson aren't we're already using `DDG.duckbar()` to render the existing auto-templated Goodies?

Wondering if we can also use your `@calls_goodie` for these Goodie's as well? That way we can let the frontend deal with rendering rather than us shoving it into the page, which causes some problems (i.e #126)